src/test_JS_repo_lib.py

```diff
@@ -1,471 +1,551 @@
-		test_info.compute_test_infras()
-		test_info.compute_nested_test_commands( test_scripts)
-		test_info.compute_test_stats()
-		# print( test_info[t])
-		# print( get_test_info(error, output))
-		test_json_summary[t] = test_info.get_json_rep()
-	return( retcode, test_json_summary)
-
-def called_in_command( str_comm, command, manager):
-	# command ends with command terminator (this list includes \0 end-of-string, 
-	# but this is not available to check in Python so we use endswith)
-	post_command_chars = [ "" ] if command.endswith(str_comm) else [ " ", "\t", ";"]
-	for pcc in post_command_chars:
-		check_comm = str_comm + pcc
-		if command.find( check_comm) == 0:
-			return( True)
-		if command.find( "&&" + check_comm) > -1 or command.find( "&& " + check_comm) > -1:
-			return( True)
-		if command.find( "cross-env NODE_ENV=test " + check_comm) > -1 or command.find( "cross-env NODE_ENV=production " + check_comm) > -1:
-			return( True)
-		if command.find( "cross-env CI=true " + check_comm) > -1:
-			return( True)
-		if command.find( "opener " + check_comm) > -1:
-			return( True)
-		if command.find( "gulp " + check_comm) > -1:
-			return( True)
-		if command.find( "nyc " + check_comm) > -1:
-			return( True)
-	return( False)
-
-def test_cond_count( test_output, regex_fct, condition, offset):
-	ptrn = re.compile( regex_fct(condition), re.MULTILINE)
-	results = ptrn.findall( test_output)
-	if offset is None:
-		return( len( results)) # just count the number of hits, each hit is an individual test (example: tap "ok" vs "not ok")
-	num_cond = 0
-	for r in results:
-		temp = r.split()
-		try:
-			num_cond += int( temp[temp.index(condition) + offset])  
-		except ValueError:
-			num_cond += 0
-	return( num_cond)
+
+
+def get_fixDeps():
+    fix_deps = []
+    for d in os.listdir("node_modules"):
+        if d[0] == '.':
+            continue
+        if d[0] == '@':
+            subFolder = os.path.join("node_modules/", d)
+            for f in os.listdir(subFolder):
+                fix_deps.append(d + '/' + f)
+
+        else:
+            fix_deps.append(d)
+
+    return fix_deps
+
+
+def get_dependencies(pkg_json, manager, include_dev_deps):
+    if pkg_json.get("devDependencies", None) and not include_dev_deps:
+        run_command("rm -r node_modules")
+        run_command("mv package.json TEMP_package.json_TEMP")
+        dev_deps = pkg_json["devDependencies"]
+        pkg_json["devDependencies"] = {}
+        with open("package.json", 'w') as f:
+            json.dump(pkg_json, f)
+            # I think there is somewrong here. "npm run install" can not install.
+            # We should use "npm install" instead.
+            # Here are some explanations:
+            # https://stackoverflow.com/questions/43664200/what-is-the-difference-between-npm-install-and-npm-run-build
+        # run_command(
+            # manager + (" install" if manager == "npm run " else ""))
+        run_command("npm" + (" install" if manager == "npm run " else ""))
```

Comment:
change this line to `run_command( "npm install" if manager == "npm run " else manager)`

You were right, the issue was a typo in that it should either reinstall dependencies with `npm install` or with `yarn`. With the existing code it was running `npm run install` which would break.